server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -274,8 +277,33 @@ public TermsEnum getTerms(boolean caseInsensitive, String string, SearchExecutio
             a = MinimizationOperations.minimize(a, Integer.MAX_VALUE);
 
             CompiledAutomaton automaton = new CompiledAutomaton(a);
-            return automaton.getTermsEnum(terms);            
-        }
+            
+            if (automaton.type == AUTOMATON_TYPE.ALL) {
+                TermsEnum result = terms.iterator();
+                if (searchAfter != null) {
+                    result = new SearchAfterTermsEnum(result, searchAfter);
+                }
+                return result;
+            }
+            return terms.intersect(automaton, searchAfter);
+        }
+        
+        // Initialises with a seek to a given term but excludes that term
+        // from results
+        public final class SearchAfterTermsEnum extends FilteredTermsEnum {
+            private final BytesRef afterRef;
+
+            public SearchAfterTermsEnum(TermsEnum tenum, BytesRef termText) {
+                super(tenum);
+                afterRef = termText;
+                setInitialSeekTerm(termText);
```

Comment:
I think you can simplify this a bit by just calling next() directly after `setInitialSeekTerm` here?

Reply:
Will try it. This class was adapted from SingleTermsEnum which differs only in that the accept method returns NO if term is not equal.

Reply:
Hmm. The test failed. The reason that led me to adding this class was that seek implementations tend to be "on or after" the provided term so you only want to call next if you're positioned _on_ the provided term. It doesn't work if the provided seek term is missing from the index (something we can expect when paging across multiple indices/shards).

Reply:
Ah yes, good point.  Maybe worth adding a comment in explaining this?